### PR TITLE
Update to rc02 smlist tests

### PR DESCRIPTION
**Pull request by sebbader-sap**
This PR introduces SubmodelElementLists into the AASFull test elements.

### CODE REVIEW

**Comment by sebbader-sap** [comment]
This PR is related to #13

File: dataformat-json/src/test/resources/test_demo_full_example.json [hunk 1]
```diff
@@ -1438,11 +1452,14 @@
...
                                 "type": "GlobalReference"
                             },
-                    "idShort": "ExampleBasicEvent"
+                    "idShort": "ExampleBasicEvent",
+                    "direction": "Output",
+                    "state": "Off"
                 },
                 {
-                    "modelType": "SubmodelElementCollection",
+                    "modelType": "SubmodelElementList",
```
Comment (mjacoby) [inline dataformat-json/src/test/resources/test_demo_full_example.json:1460]: Changing element type also requires to update related elements in whole document, e.g. references such as https://github.com/eclipse-digitaltwin/aas4j/blob/9e50db894089ec938f457c9d2b5acc6e59e7f082/dataformat-json/src/test/resources/test_demo_full_example.json#L715

**Review by mjacoby — CHANGES_REQUESTED** [review]
Changes in AAS full example introduced inconcsistencies that need to be fixed.

File: dataformat-json/src/test/resources/test_demo_full_example.json [hunk 2]
```diff
@@ -270,6 +270,17 @@
                     "type": "GlobalReference"
                 }
             ],
+            "dataSpecifications": [
+                {
+                    "keys": [
+                        {
+                            "type": "GlobalReference",
```
[the remark this answers is not in the document]
Comment (sebbader) [reply dataformat-json/src/test/resources/test_demo_full_example.json:277]: It's somehow redundant - but yes, that's the official pattern for V3.0RC02.

File: dataformat-core/src/test/java/org/eclipse/aas4j/v3/dataformat/core/AASFull.java [hunk 3]
```diff
@@ -1546,9 +1542,10 @@ public static Submodel createSubmodel7() {
                                 .type(ReferenceTypes.GLOBAL_REFERENCE)
                                 .build())
                         .build())
-                .submodelElements(new DefaultSubmodelElementCollection.Builder()
+                .submodelElements(new DefaultSubmodelElementList.Builder()
```
[the remark this answers is not in the document]
Comment (sebbader) [reply dataformat-core/src/test/java/org/eclipse/aas4j/v3/dataformat/core/AASFull.java:1582]: Valid point, just changed this with todays's commits.

File: dataformat-json/src/test/java/org/eclipse/aas4j/v3/dataformat/json/JsonReferableDeserializerTest.java [hunk 4]
```diff
@@ -109,12 +105,23 @@ public void testReadSubmodelElements() throws IOException, DeserializationExcept
...
+        String expected = Files.readString(fileExpected.toPath());
+        SubmodelElementList submodelElementList = new JsonDeserializer().readReferable(expected, SubmodelElementList.class);
+        Environment environment = AASFull.ENVIRONMENT;
+        SubmodelElement submodelElementListExpected = environment.getSubmodels().get(6).getSubmodelElements().get(5);                ;
+
+        assertEquals(submodelElementListExpected, submodelElementList);
+    }
+
+    @Test
+    public void testReadSubmodelElementColelction() throws IOException, DeserializationException {
```
[the remark this answers is not in the document]
Comment (sebbader) [reply dataformat-json/src/test/java/org/eclipse/aas4j/v3/dataformat/json/JsonReferableDeserializerTest.java]: done

File: dataformat-json/src/test/java/org/eclipse/aas4j/v3/dataformat/json/JsonReferableSerializerTest.java [hunk 5]
```diff
@@ -79,14 +79,14 @@ public void testSerializeSubmodelelements() throws IOException, SerializationExc
         Environment environment = AASFull.ENVIRONMENT;
         SubmodelElement submodelElement0 = environment.getSubmodels().get(0).getSubmodelElements().get(0);
         SubmodelElement submodelElement1 = environment.getSubmodels().get(0).getSubmodelElements().get(1);
-        compare("src/test/resources/submodelElementList.json",submodelElement0,submodelElement1);
+        compare("src/test/resources/listOfSubmodelElements.json",submodelElement0,submodelElement1);
     }
 
     @Test
-    public void testSerializeSubmodelelementCollection() throws IOException, SerializationException, JSONException {
+    public void testSerializeSubmodelelementList() throws IOException, SerializationException, JSONException {
```
[the remark this answers is not in the document]
Comment (sebbader) [reply dataformat-json/src/test/java/org/eclipse/aas4j/v3/dataformat/json/JsonReferableSerializerTest.java]: added a SMC UnitTest again

**Comment by sebbader** [comment]
@mjacoby please check the PR again, I hope I have regarded every point.

File: dataformat-json/src/test/java/org/eclipse/aas4j/v3/dataformat/json/JsonReferableDeserializerTest.java [hunk 6]
```diff
@@ -24,16 +24,12 @@
 import java.util.List;
 
 import org.eclipse.aas4j.v3.dataformat.core.AASFull;
+import org.eclipse.aas4j.v3.model.*;
```
[the remark this answers is not in the document]
Comment (mjacoby) [reply dataformat-json/src/test/java/org/eclipse/aas4j/v3/dataformat/json/JsonReferableDeserializerTest.java]: Still unresolved...looks like it was fixed with one commit and then re-introduced in later commit.

[the remark this answers is not in the document]
Comment (sebbader) [reply dataformat-json/src/test/java/org/eclipse/aas4j/v3/dataformat/json/JsonReferableDeserializerTest.java]: Damn. My Intelij replaces the imports by it's own if a certain number of class imports happened. Need to deactivate this feature.

[the remark this answers is not in the document]
Comment (sebbader) [reply dataformat-json/src/test/java/org/eclipse/aas4j/v3/dataformat/json/JsonReferableDeserializerTest.java]: @mjacoby please check again.

**Comment by mjacoby** [comment]
**Current stauts**

Problems with handling of default values, e.g. for 'SubmodelElementList.orderRelevant'. After discussion with @sebbader-sap we decided that this be solved on model level, i.e. default values should be set in constructors of default implementation classes. This requires updating the generator but also the RDF file to contain information about default values (which it currently does not). Not sure if RDF file is currently auto-generated by another tool, but if so, this might require to update this tool too. 

@sebbader-sap is working on this issue
@mjacoby will review changes once this is done

**Comment by sebbader-sap** [comment]
@mjacoby please comment here after we have finished https://github.com/eclipse-digitaltwin/aas4j-model-generator/issues/6 ?

**Comment by sebbader-sap** [comment]
Then I will resolve the conflicts and we can merge the PR.

**Pull request merged**